src/rust-crypto/DehydratedDeviceManager.ts

```diff
@@ -267,12 +283,14 @@ export class DehydratedDeviceManager {
      * Creates and stores a new key in secret storage if none is available.
      */
     public async createAndUploadDehydratedDevice(): Promise<void> {
-        const key = (await this.getKey(true))!;
+        const key = ((await this.getCachedKey()) || (await this.getKey(true)))!;
```

Comment:
```suggestion
        const key = (await this.getKey(true))!;
```
since `getKey` already calls `getCachedKey`